main.py

```diff
@@ -23,14 +25,23 @@ def check(a, b):
     and if so what type it will be"""
     state = Board[a][b].CurrentState
     total = [0, 0]
-    total[Board[a + 1][b].CurrentState] += 1
-    total[Board[a][b + 1].CurrentState] += 1
-    total[Board[a - 1][b].CurrentState] += 1
-    total[Board[a][b - 1].CurrentState] += 1
-    total[Board[a + 1][b + 1].CurrentState] += 1
-    total[Board[a - 1][b - 1].CurrentState] += 1
-    total[Board[a + 1][b - 1].CurrentState] += 1
-    total[Board[a - 1][b + 1].CurrentState] += 1
+    aa = a - 1
+    ab = a + 1
+    ba = b - 1
+    bb = b + 1
+    if config.Wrap and a == config.Width-1:
+        ab = 0
+    if config.Wrap and b == config.Height-1:
+        bb = 0
+
+    total[Board[aa][b].CurrentState] += 1
+    total[Board[a][ba].CurrentState] += 1
+    total[Board[ab][b].CurrentState] += 1
+    total[Board[a][bb].CurrentState] += 1
+    total[Board[aa][ba].CurrentState] += 1
```

Comment:
what is `ab`? what is `bb`? you should make a loop here

Reply:
the first letter tells you which direction on the board the variable relates to, the second tells you whether it's to the left/up (a) or to the right/down (b), I will try and find a more clear way of setting these out

Reply:
renamed them (sort of)